tests/sites_tests/tests.py

```diff
@@ -135,6 +135,12 @@ def test_clear_site_cache_domain(self):
         clear_site_cache(Site, instance=site, using='default')
         self.assertEqual(models.SITE_CACHE, {})
 
+    def test_unique_domain(self):
+        site = Site(domain=self.site.domain)
+        msg = "{'domain': ['Site with this Domain name already exists.']}"
```

Comment:
Just test with `'Site with this Domain name already exists.'`. This might fail on Python 2 because of unicode literals representation.